### PR TITLE
Iiif3 downloads

### LINKED ISSUE #8696: Refactor downloadOptions/Viewer code
Relates to #8608

Refactor code related to download options on works and items pages.

If a work has a iiif-presentation manifest for a DigitalLocation we get the download options directly from that. If a work has a iiif-image for a DigitalLocation we create the download links from the location information. However, we then fetch the image json from the iiif image server client side, so we know the dimensions of the image and update the UI accordingly. If a work has a iiif-presentation and iiif-image location, use the iiif-presentation location.

TODO:
- create and use fetch function for the image json
- create and use transform for the image json
- create missing types
- create and use useIIIFImage hook
- make sure we give precedence to iiif-presentation location over iiif-image location
- Pass either TransformedManifest or TransformedImage to the IIIFViewer, not both

### PR DESCRIPTION
**Pull request by gestchild**
Fixes #8696 

Some refactoring of how we create download links, before moving it over to use iiif3 in #8697

- create and uses fetch and transform functions for iiif-image json
- removes some duplicate props on the TransformedManifest
- adds some comments which hopefully makes it clearer how the download links are created

### CODE REVIEW

File: catalogue/webapp/pages/download.tsx [hunk 1]
```diff
@@ -27,29 +21,30 @@ import { GetServerSideProps, NextPage } from 'next';
...
+    transformedManifest;
+  const displayTitle = title || work?.title || '';
   const iiifImageLocation = work
     ? getDigitalLocationOfType(work, 'iiif-image')
     : null;
   const iiifPresentationLocation = work
     ? getDigitalLocationOfType(work, 'iiif-presentation')
     : null;
-  const digitalLocation = iiifImageLocation || iiifPresentationLocation;
+  const digitalLocation = iiifImageLocation || iiifPresentationLocation; // TODO here we favour imageLocation over manifestLocation
```
Comment (alexwlchan) [inline catalogue/webapp/pages/download.tsx:47]: What's this TODO telling us? That in future we want to prefer manifest location? Or some other change?

File: catalogue/webapp/pages/download.tsx [hunk 2]
```diff
@@ -58,35 +53,24 @@ const DownloadPage: NextPage<Props> = ({
...
-    ? isUiEnabled(getUiExtensions(manifest), 'mediaDownload')
-    : true;
+  const credit = (iiifImageLocation && iiifImageLocation.credit) || iiifCredit;
 
   return (
     <PageLayout
-      title={title}
+      title={displayTitle}
       description=""
-      url={{ pathname: `/works/${workId}/download`, query: { sierraId } }}
```
Comment (alexwlchan) [inline catalogue/webapp/pages/download.tsx:89]: Do we use this Sierra ID for tracking?

**Review by alexwlchan — APPROVED** [review]
🍨 🧊 ⛸️ 3️⃣

File: catalogue/webapp/components/Download/Download.tsx [hunk 3]
```diff
@@ -97,6 +97,9 @@ const Download: NextPage<Props> = ({
 }: Props) => {
   const downloadsContainer = useRef(null);
   const { isEnhanced } = useContext(AppContext);
+  const downloadOptionsWithoutText = downloadOptions.filter(
+    option => option.format !== 'text/plain'
+  );
```
[the remark this answers is not in the document]
Comment (gestchild) [reply catalogue/webapp/components/Download/Download.tsx:102]: I'm going to move this to the transformer in the next PR and will add a comment there

File: catalogue/webapp/components/Download/Download.tsx [hunk 4]
```diff
@@ -118,37 +121,35 @@ const Download: NextPage<Props> = ({
...
-                    .map(option => {
-                      const action = option['@id'].match(/\/full\/full\//)
-                        ? 'download large work image'
-                        : option['@id'].match(/\/full\/760/)
-                        ? 'download small work image'
-                        : option.label;
-                      const format = getFormatString(option.format);
+                  {downloadOptionsWithoutText.map(option => {
+                    const action = option['@id'].match(/\/full\/full\//)
+                      ? 'download large work image'
```
[the remark this answers is not in the document]
Comment (gestchild) [reply catalogue/webapp/components/Download/Download.tsx:126]: I'll raise the with Alice and Dom, but won't do it in this PR

**Pull request merged**